java/test/jmri/jmrit/display/layoutEditor/LayoutEditorToolsTest.java

```diff
@@ -487,92 +1116,128 @@ public void testFinalizeBlockBossLogicNullInput() {
         Assert.assertFalse("Signal head not on panel after remove", let.isHeadOnPanel(signalHeads.get(1)));
     }
 
     @Test
-    @Ignore("Consistently fails on AppVeyor and Windows 12/20/2019")
+    @Ignore("Fails on AppVeyor and Windows 12/20/2019")
     public void testSetSignalHeadOnPanelAtXYDoubleAndRemove() {
         Assume.assumeFalse(GraphicsEnvironment.isHeadless());
         Assert.assertFalse("Signal head not on panel before set", let.isHeadOnPanel(signalHeads.get(1)));
         let.setSignalHeadOnPanel(0.D, "IH1", 0, 0);
         //setSignalHeadOnPanel performs some GUI actions, so give
         //the AWT queue some time to clear.
-        new QueueTool().waitEmpty(100);
+        new EventTool().waitNoEvent(100);
         Assert.assertTrue("Signal head on panel after set", let.isHeadOnPanel(signalHeads.get(1)));
         let.removeSignalHeadFromPanel("IH1");
         //removeSignalHeadFromPanel performs some GUI actions, so give
         //the AWT queue some time to clear.
-        new QueueTool().waitEmpty(100);
+        new EventTool().waitNoEvent(100);
         Assert.assertFalse("Signal head not on panel after remove", let.isHeadOnPanel(signalHeads.get(1)));
     }
 
     @Test
-    @Ignore("causes error on jenkins; exhausts failure retries")
+    ///@Ignore("Fails on AppVeyor, macOS and Windows 12/20/2019")
     public void testGetSignalHeadIcon() {
         Assume.assumeFalse(GraphicsEnvironment.isHeadless());
-        Assert.assertNotNull("Signal head icon for panel", let.getSignalHeadIcon("IH1"));
+        Assert.assertNotNull("let null", let);
+        for (SignalHead sh : signalHeads) {
+            Assert.assertNotNull("Signal head icon for panel", let.getSignalHeadIcon(sh.getSystemName()));
+        }
     }
 
     @Test
     public void testIsHeadOnPanel() {
         Assume.assumeFalse(GraphicsEnvironment.isHeadless());
-        Assert.assertFalse("Signal head not on panel", let.isHeadOnPanel(signalHeads.get(1)));
+        for (SignalHead sh : signalHeads) {
+            Assert.assertFalse("Signal head not on panel", let.isHeadOnPanel(sh));
+        }
     }
 
     @Test
     public void testIsHeadAssignedAnywhere() {
         Assume.assumeFalse(GraphicsEnvironment.isHeadless());
-        Assert.assertFalse("Signal head not on panel", let.isHeadAssignedAnywhere(signalHeads.get(1)));
+        for (SignalHead sh : signalHeads) {
+            Assert.assertFalse("Signal head not on panel", let.isHeadAssignedAnywhere(sh));
+        }
     }
 
     @Test
     public void testRemoveSignalHeadAssignment() {
         Assume.assumeFalse(GraphicsEnvironment.isHeadless());
         //just verify this doesn't thrown an error.
-        let.removeAssignment(signalHeads.get(1));
+        for (SignalHead sh : signalHeads) {
+            let.removeAssignment(sh);
+        }
     }
 
     @Test
-    @Ignore("causes error on jenkins; exhausts failure retries")
+    ///@Ignore("Fails on AppVeyor, macOS and Windows 12/20/2019")
```

Comment:
There isn't any change here other than commenting out the ignore annotation.  What changed to make this work reliably?